### DISCUSSION TITLE
Wiring Paths / Auto-management / Autorouting

### DISCUSSION THREAD

**Discussion by Saevon** [FakeFishGames/Barotrauma#4713] [Ideas and suggestions]
# Problem

Wiring is very tedious, mostly due to wire cleanliness being hard to do. Also when moving machinery around there is a lot of wire cleaning required.

When editing an existing sub, the problem is even stronger, becoming a right mess unless you're very careful, and even then its quite a pain. So being able to attach a wire to a cable-path would also make sub-customizations nicer!

# Solution

Linking a wire to a "Wire Path" with a "click" would automatically follow it. Then the wire exits at the closest point to the next Wire-Node (if this is machinery, wiring becomes really easy. If its a new node, you still retain control to create custom paths)

Wire paths could then be linked to allow any the wires to go around corners, and get to any part of the ship easily. (Or they could automatically transfer between overlapped ones)

The items that would work best are:

 * Cable Holders
 * Vent Pipe A
 * Decorative wire
 *  Decorative Column
 * Ladder
 * Door
 * Vertical Back (Small/Large)

These items would also have an option for "Display wires" (default ON only for the Cable Holders).   In the wiring view the paths show up regardless when they follow a wire path.

P.S. Vent Pipe A variants should have "Vent Pipe with Wires" in its name.

**Comment by jpjpjp838** [comment]
It's sound so helpfull for us who keep remodeling our sub! Thanks for the answer!!

**Comment by Arctodites** [comment]
This sounds like a great idea.  If one could simply lay down pre-traced wiring paths to which other wires would snap, instead of having to nudge individual signal cables 4 ticks to the left of a grid point where power cables are snapped ... that would be hugely more efficient.

It would be helpful if the autorouting feature also allowed us to specify a sprite depth for all wires along that path, because as it stands, sprite depth levels in the Sub Editor are wildly inconsistent, contradictory, and in many cases plain wrong (as in, bugged -- but that is the subject of an upcoming bug report I'm drafting rn).  That makes it very hard to know whether wires will appear in front or behind other sprites, and leads to countless minute layer adjustments that one has to totally redo when it turns out that other sprites won't draw the way you planned.

**Reply by Saevon** [reply]
I figure the wire would always be  displayed at "cable-holder - 10" sprite depth. So it can be changed however you want.  But any wires that come OUT or between two cable holders would probably use the default ".850" that all wires have (aka the sprite-depth the wire itself has.
